### ISSUE TITLE
[class-literal-property-style] should not fix readonly to getter when assigned in ctor

### ISSUE THREAD

**Issue by k3nsei** [typescript-eslint/typescript-eslint#2206]
**Repro**

[Playground](https://typescript-eslint.io/play#ts=5.3.2&fileType=.tsx&code=MYGwhgzhAECC0G8BQ1oAcBOBLAbmALgKbQaFgAmA9gHYgCe0AZpZQFzQT7bUDm0AvNADkAIzAYhAbiQpowGpwwBXYPkoYAFMzYcuWXgEpEs1PgAWWCADptApi2moAvkhdA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6M%2BAQ2WVsf0TRO8WsWgB7UtCK1kRJOjABtcNhyCJ0SABpVayAHNE%2BAdFR6wAXV36CJclRr1kjFqzGIAZoNrREnABNxJnhCRUgAd05oJkhVAF8QeKA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)

```JSON
{
  "rules": {
    "@typescript-eslint/class-literal-property-style": ["error", "getters"],
    "@typescript-eslint/prefer-readonly": "warn"
  }
}
```

```TS
export class A {
  private readonly foo: string = 'bar';

  constructor(foo: string) {
    this.foo = foo;
  }
}
```

**Expected Result**
Code should be unchanged

**Actual Result**
Code after fix is invalid
```TS
export class A {
  private get foo() { return 'bar'; }

  constructor(foo: string) {
    this.foo = foo;
  }
}
```

**Additional Info**
None

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.2.0` |
| `@typescript-eslint/parser`        | `3.2.0` |
| `TypeScript`                       | `3.9.5` |
| `ESLint`                           | `7.2.0` |
| `node`                             | `14.2.0` |
| `npm`                              | `6.14.5` |
| `yarn`                              | `1.22.4` |

**Comment by k3nsei** [comment]
@all-contributors please add k3nsei for bug

**Comment by JBhagat841** [comment]
Is anyone currently working on this? Can I take a stab at it?

**Comment by bradzacher** [comment]
feel free to have a go!

**Comment by JBhagat841** [comment]
@JoshuaKGoldberg wanna pair on this?

**Comment by JoshuaKGoldberg** [comment]
![Funny looking kid dancing to "Hell Yeah!"](https://user-images.githubusercontent.com/3335181/85027349-d0143100-b147-11ea-9dfc-22405c216311.gif)

**Comment by jerone** [comment]
[Playground](https://typescript-eslint.io/play/#ts=5.0.4&fileType=.tsx&code=KYDwDg9gTgLgBAYwDYEMDOa4EE4G8CwAUHHGFAJYBuKMwcUwKAJhAHZICecAZhBAFxw0MCqwDmcALxwA5ACMUUGQG4iREgjbCoAVwQxoACl4ChI8uICUedSTgwAFuTQA6E1J59VxO5tZoIJGAXJAgxQ0dnNz5LbxIAXyJ4oA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6M%2BAQ2WVsf0TRO8WsWgB7UtCK1kRJOjABtcNhyCJ0SABpVayAHNE%2BAdFR6wAXV36CJclRr1kjFqzGIAZoNrREnABNxJnhCRUgAd05oJkhVAF8QeKA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false).

Error:
> Literals should be exposed using getters. 2:20 - 2:23
